src/lib/viewers/doc/DocBaseViewer.js

```diff
@@ -167,6 +172,11 @@ class DocBaseViewer extends BaseViewer {
             this.pageControls.removeListener('pagechange', this.setPage);
         }
 
+        if (this.zoomControls) {
+            this.zoomControls.removeListener(ZOOM_CONTROLS_EVENTS.zoomin, this.zoomIn);
```

Comment:
It seems like logic should live in `destroy` method on the ZoomControls class, which should then be called from the `this.controls.destroy()` method.